lib/src/protocol/kawa_h1/editor.rs

```diff
@@ -0,0 +1,197 @@
+                    "{}, {}",
+                    unsafe { from_utf8_unchecked(header.val.data(buf)) },
+                    peer_ip.to_string()
+                ));
+            }
+            if let Some(header) = &mut forwarded {
+                let value = unsafe { from_utf8_unchecked(header.val.data(buf)) };
+                let new_value = match (peer_ip, public_ip) {
+                    (IpAddr::V4(_), IpAddr::V4(_)) => {
+                        format!("{value}, proto={proto};for={peer_ip}:{peer_port};by={public_ip}")
+                    }
+                    (IpAddr::V4(_), IpAddr::V6(_)) => {
+                        format!(
+                            "{value}, proto={proto};for={peer_ip}:{peer_port};by=\"{public_ip}\""
+                        )
+                    }
+                    (IpAddr::V6(_), IpAddr::V4(_)) => {
+                        format!(
+                            "{value}, proto={proto};for=\"{peer_ip}:{peer_port}\";by={public_ip}"
+                        )
+                    }
+                    (IpAddr::V6(_), IpAddr::V6(_)) => {
+                        format!(
+                        "{value}, proto={proto};for=\"{peer_ip}:{peer_port}\";by=\"{public_ip}\""
+                    )
+                    }
+                };
+                header.val = kawa::Store::from_string(new_value);
+            }
+
+            if !has_x_for {
+                request.push_block(kawa::Block::Header(kawa::Pair {
+                    key: kawa::Store::Static(b"X-Forwarded-For"),
+                    val: kawa::Store::from_string(peer_ip.to_string()),
+                }));
+            }
+            if !has_forwarded {
+                let value = match (peer_ip, public_ip) {
+                    (IpAddr::V4(_), IpAddr::V4(_)) => {
+                        format!("proto={proto};for={peer_ip}:{peer_port};by={public_ip}")
+                    }
+                    (IpAddr::V4(_), IpAddr::V6(_)) => {
+                        format!("proto={proto};for={peer_ip}:{peer_port};by=\"{public_ip}")
+                    }
+                    (IpAddr::V6(_), IpAddr::V4(_)) => {
+                        format!("proto={proto};for=\"{peer_ip}:{peer_port}\";by={public_ip}")
+                    }
+                    (IpAddr::V6(_), IpAddr::V6(_)) => {
+                        format!("proto={proto};for=\"{peer_ip}:{peer_port}\";by=\"{public_ip}\"")
+                    }
+                };
+                request.push_block(kawa::Block::Header(kawa::Pair {
+                    key: kawa::Store::Static(b"Forwarded"),
+                    val: kawa::Store::from_string(value),
+                }));
+            }
+        }
+
+        request.push_block(kawa::Block::Header(kawa::Pair {
+            key: kawa::Store::Static(b"Sozu-Id"),
+            val: kawa::Store::from_string(self.id.to_string()),
+        }));
+    }
+
+    fn on_response_headers(&mut self, response: &mut GenericHttpStream) {
```

Comment:
please add a documenting comment to explain what it does to the generic request (and to self)